keep-ui/app/settings/api-key-settings.tsx

```diff
@@ -19,29 +37,114 @@ interface Props {
-    (url) => fetcher(url, accessToken),
+    selectedTab === "api-key" ? `${apiUrl}/settings/apikeys` : null,
+    async (url) => {
+        const response = await fetcher(url, accessToken);
+        setApiKeys(response.apiKeys);
+        return response;
+    },
     { revalidateOnFocus: false }
   );
 
+
+  const { data: configData } = useSWR<Config>("/api/config", fetcher, {
+    revalidateOnFocus: false,
+  });
+
+
+  const [isApiKeyModalOpen, setApiKeyModalOpen] = useState(false);
+  const [apiKeys, setApiKeys] = useState([]);
+
   if (isLoading) return <Loading />;
   if (error) return <div>{error.message}</div>;
 
-  const copyBlockApiKeyProps = {
-    theme: { ...a11yLight },
-    language: "text",
-    text: data?.apiKey || "",
-    codeBlock: true,
-    showLineNumbers: false,
-  };
+  const getCopyBlockProps = (secret: string) => {
+    return {
+        theme: { ...a11yLight },
+        language: "text",
+        text: secret,
+        codeBlock: true,
+        showLineNumbers: false,
+      }; 
+  }
+
+  // Determine runtime configuration
+  const authType = configData?.AUTH_TYPE as AuthenticationType;
+  // Create API key disabled if authType is none
+  const createApiKeyEnabled = authType !== AuthenticationType.NO_AUTH
 
   return (
     <div className="mt-10">
-      <Title>API Key</Title>
+      <div className="flex justify-between">
+        <div className="flex flex-col">
+          <Title>API Keys</Title>
+          <Subtitle>Add, edit or deactivate API keys from your tenant</Subtitle>
+        </div>
+
+        <div>
+          <Button
+            color="orange"
+            size="md"
+            icon={KeyIcon}
+            onClick={() => setApiKeyModalOpen(true)}
+            disabled={!createApiKeyEnabled}
+            tooltip={!createApiKeyEnabled ? "API Key creation is disabled because Keep is running in NO_AUTH mode.": "Add user"}
+          >
+            Create API key 
+          </Button>
+        </div>
+      </div>
       <Card className="mt-2.5">
+       {apiKeys.length ? (
+        <Table>
+          <TableHead>
+            <TableRow>
+              <TableHeaderCell>Name</TableHeaderCell>
+              <TableHeaderCell className="text-right">Key</TableHeaderCell>
+              <TableHeaderCell className="text-right">
+                Created At
+              </TableHeaderCell>
+              <TableHeaderCell className="text-right">
+                Last Used 
+              </TableHeaderCell>
+              <TableHeaderCell>
+              {/* Menu */}
+              </TableHeaderCell>
+            </TableRow>
+          </TableHead>
+          <TableBody>
+            {apiKeys.map((key) => (
+              <TableRow
+                key={key.reference_id}
+              >
+                <TableCell>{key.reference_id}</TableCell>
+                <TableCell className="text-right">
+                  <CopyBlock {...getCopyBlockProps(key.secret)} />
```

Comment:
There is a little issue when I'm clicking on "copy" on more than on API key (only one should be shown as copied):
![CleanShot 2024-01-29 at 14 25 51](https://github.com/keephq/keep/assets/12069200/6dc2c0fa-8f1e-4e71-9910-cd417997481f)

Reply:
CopyBlock component [handles state internally](https://github.com/rajinwonderland/react-code-blocks/blob/master/packages/react-code-blocks/src/components/CopyBlock.tsx#L80).